Framework/MDAlgorithms/src/MDNorm.cpp

```diff
@@ -365,6 +365,32 @@ std::map<std::string, std::string> MDNorm::validateInputs() {
       }
     }
   }
+  // validate accumulation workspaces, if provided
+  // this only checks that, if given, the two workspaces
+  // have the same dimensions.  We verify that binning parameters
+  // are the same when we execute the algorithm.
+  boost::shared_ptr<IMDHistoWorkspace> tempNormWS =
+      this->getProperty("TemporaryNormalizationWorkspace");
+  Mantid::API::IMDHistoWorkspace_sptr tempDataWS =
+      this->getProperty("TemporaryDataWorkspace");
+  if (tempNormWS || tempDataWS) {
+    if ((!tempNormWS) || (!tempDataWS)) {
```

Comment:
This should be replaced with `tempNormWS && tempDataWS`. Then do the for loop. The error message should follow an `else`. Currently you are likely to dereference a null pointer. Not good.